Detectors/Calibration/include/DetectorsCalibration/Utils.h

```diff
@@ -25,18 +25,65 @@
 #include "CCDB/CcdbObjectInfo.h"
 #include "CommonUtils/MemFileHelper.h"
 #include "CCDB/CcdbApi.h"
+#include <vector>
 
 namespace o2
 {
 namespace calibration
 {
 
 struct Utils {
+
+  enum ValueType { Invalid = 0,
+                   Interpolation,
+                   ClosestAvailableFromBelow,
+                   ClosestAvailableFromAbove,
+                   SameAsRequested };
+
   static constexpr o2::header::DataOrigin gDataOriginCDBPayload{"CLP"}; // generic DataOrigin for calibrations payload
   static constexpr o2::header::DataOrigin gDataOriginCDBWrapper{"CLW"}; // generic DataOrigin for calibrations wrapper
   template <typename T>
   static void prepareCCDBobjectInfo(T& obj, o2::ccdb::CcdbObjectInfo& info, const std::string& path,
                                     const std::map<std::string, std::string>& md, long start, long end = -1);
+
+  static std::pair<int, double> findPair(std::vector<std::pair<uint64_t, double>> vect, uint64_t timestamp)
+  {
+    // function to find the pair in the vector with the timestamp closest in the past to the one passed as argument
+    // if two entries in the vector have the same timestamp, the first that is found is used
+    std::vector<uint64_t> times;
+    for (const auto& el : vect) {
+      times.push_back(el.first);
+    }
+    auto lower = std::lower_bound(times.begin(), times.end(), timestamp);
```

Comment:
use lambda instead of creating a vector:
```
auto lower = std::lower_bound(times.begin(), times.end(), timestamp, [](const std::pair<uint64_t, double>& p, uint64_t value) { return p.first < value; } );
```

Respectivelt, use directly the `vect` instead of `times` below.

Reply:
Done.